mlflow/utils/autologging_utils/__init__.py

```diff
@@ -538,16 +538,21 @@ def __enter__(self):
         def __exit__(self, tp, val, traceback):
             _TrainingSession._session_stack.pop()
 
-        def should_log(self):
-            """
-            Returns True when at least one of the following conditions satisfies:
+        def is_root(self):
+            return self._parent is None
 
-            1. This session is the root session.
-            2. The parent session allows autologging and its class differs from this session's
-               class.
+        def should_use_patch(self):
+            """
+            Indicates whether we should use a patch implementation in this session.
             """
-            return (self._parent is None) or (
-                self._parent.allow_children and self._parent.clazz != self.clazz
+            return self.is_root() or (
+                self._parent.allow_children
+                # For lightgbm autologging, we patch both `LGBMClassifier.fit` and `LGBMModel.fit`.
+                # Since `LGBMClassifier` is a subclass of `LGBMModel`,
+                # `LGBMClassifier.fit` creates two sessions, one for `LGBMClassifier.fit` and one
+                # for `LGBMModel.fit`. To prevent `LGBMModel.fit` from using the patch
+                # implementation, check the identity of the estimator object.
+                and self._parent.estimator is not self.estimator
```

Comment:
@WeichenXu123 Is there an issue if we have multiple sessions on the same estimator object?